pkg/jx/cmd/shell.go

```diff
@@ -177,12 +176,22 @@ func (o *ShellOptions) Run() error {
 		env = append(env, fmt.Sprintf("ZDOTDIR=%s", tmpDirName))
 		e = exec.Command(shell, "-i")
 		e.Env = env
+	} else if shell == "cmd.exe" {
+		env :=os.Environ()
+		env = append(env, fmt.Sprintf("KUBECONFIG=%s", tmpConfigFileName))
+		e = exec.Command(shell)
+		e.Env = env
 	}
 
 	e.Stdout = o.Out
 	e.Stderr = o.Err
 	e.Stdin = os.Stdin
-	return e.Run()
+	err = e.Run()
+	if deleteError := os.RemoveAll(tmpDirName); deleteError != nil {
```

Comment:
moved the cleanup to here, and only clean up the directory we just created to not effect other shells.